### ISSUE TITLE
Ethos Suite USB connection

### ISSUE THREAD

**Issue by Fin35GH** [FrSkyRC/ETHOS-Feedback-Community#1719]
After connecting my X20S Tx in Bootloader mode to my PC, to interact with the Ethos Suite app, I find that the two Tx drives disconnect periodically and Windows eventually reports a USB device error.
This is the error I see in Windows Device Manager:
“Windows has stopped this device because it has reported problems. (Code 43)
A request for the USB device descriptor failed.”
 
Other checks:
•	It’s fine if I don’t run the app, ie the two USB drives stay connected and their folders and files can be seen in File Explorer.
 
•	Ethos Suite seems to work successfully in Power Off mode and I successfully installed the latest Bootloader via the DFU option in Ethos Suite (after installing a suitable driver: “Windows Installer reconfigured the product. Product Name: DfuSe v3.0.6. Product Version: 3.0.6. Product Language: 1033. Manufacturer: STMicroelectronics. Reconfiguration success or error status: 1602”)
 
•	I don’t have any problems with other USB devices connected to my PC via the same port.
 
So this seems to imply a bug in Ethos Suite. Anybody else having issues?
 
My configuration:
Windows 10 Pro with auto updates (05/06/2022), Version 21H1, OS build 19043.1706
USB 3.0 port on PC
FrSky Bootloader: 1.2.10
FrSky Firmware: 1.2.10 EU
Ethos Suite 1.2.10

**Comment by bsongis-frsky** [comment]
Would you please try:
- a different port
- a different cable

**Comment by Fin35GH** [comment]
Thanks for this suggestion.

I had previously tried a different USB port and I have just tried a different cable (which works correctly with other devices), but the result is the same: with the Tx in Bootloader mode, both Tx drives show correctly in File Explorer until I start Ethos Suite, at which point Windows reports that the drives are unrecognized and they disappear from File Explorer.

If I connect the Tx to the USB whilst Ethos Suite is already running, it shows the X20S menu item intermittently but eventually it gives up and the Tx drives become unrecognized by Windows.

**Comment by Fin35GH** [comment]
One other observation: with the Tx connected in Power Off mode, Ethos Suite does not show the DFU option if it is started after the cable is connected, ie Ethos Suite must be running before the cable is connected (have I missed this in the instructions?).

**Comment by bsongis-frsky** [comment]
Strange indeed, is it the same on another computer?
Have you tried USB2.0?

**Comment by bsongis-frsky** [comment]
I think that your computer detects that the radio uses too much current, then it disconnects and reconnects one second later, then disconnects for the same reason ec.

**Comment by Fin35GH** [comment]
That is exactly what I see! My motherboard has two USB sites, and I've just tried the alternative one, but it has the same issue. I will purchase a powered USB hub to see if that solves the problem. Do you know what current is required?
Thanks.

**Comment by marciomc-github** [comment]
Many colleagues who recently bought their X20 related the same issue. We found out it is probably related to the usb type c cable provided by FrSky. 
Changed the cable and problem was solved. 
It seems FrSky lower the standard for the usb cable.

**Comment by Fin35GH** [comment]
Thank for the suggestion. I had previously tried a different cable that I know works, but I tried it again anyway - still no joy. 

I have also purchased a powered USB 3.0 hub and fully re-charged the Tx but the problem remains.

Given that this only happens with Ethos Suite in Bootloader mode, I would have thought this suggests a software issue, but I will try testing another PC in the next few days and report back.

**Comment by bsongis-frsky** [comment]
OK please continue posting your results, I will reopen the issue if needed

**Comment by Fin35GH** [comment]
I have just updated the BIOS and USB drives on my desktop PC with the latest ASUS versions. Now the Tx drives do not become "unrecognized" and they remain accessible in File Explorer when Ethos Suite is running with the Tx in Bootloader mode. However, the X20S menu item in Ethos Suite still flashes on and off randomly - mostly off. All this is with USB3.0 (with standard USB the X20S menu item never appears). I've tried all the usual cable swaps and external powered hub combinations but they have no effect. 

FYI: Device Manager /.../ USB Root Hub / Power / shows maximum current 500mA per port, but I'm not sure if this is relevant when using a powered external USB hub.
